IndomitableIbises/Bangazon-API-Sprint-1
# Rc customer model

## Link to Ticket
Closes #1 

## Description of Proposed Changes

Created a model for customer. Based on ERD. Created a viewset that will show all customers as well as specific customers based on id in the url path. Can add, delete and update customer instances.

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout Rc customer model
```
```
python manage.py migrate
python manage.py runserver
```

1. In the new localhost server that is made, add customers/ at the end of your url
2. There should be a page with no data listed, but a form at the bottom to input required fields
3. The three required fields are first name, last name, and a true false value of the account being active (Which is viewed as a check box)
4. account_date and last_login will populate automatically the first time a new instance is generated
5. After inputting some data, check to make sure the page reflects changes
6. After a few instances have been made, append a customer id to the url (i.e. customers/1/)
7. You should only be shown that particular instance
8. You can update the data with new information down that the bottom, press submit and test that it all updates correctly
9. Open your sqlite database after everything, and confirm all the personal changes and additions have been updated and added to the table correctly.
10. Double check that the data table is formatted correctly


## Impacted Areas in Application

List general components of the application that this PR will affect:

*bangazon/setting.py
*bangazon/urls.py
*API/models.py
*API/views.py
*API/urls.py
*API/serializers.py

## Mentions @username

@jcarter0149 @sirwin4 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
